ods_ci/libs/DataSciencePipelinesKfpTekton.py

```diff
@@ -0,0 +1,110 @@
+from robotlibcore import keyword
+import base64
+from kfp import dsl
+from kfp import components
+import kfp_tekton
+from kfp_tekton.compiler import TektonCompiler
+from DataSciencePipelinesAPI import DataSciencePipelinesAPI
+
+
+# The pipelines methods are static
+# ----------------------------------- begin pipeline methods -----------------------------------
+def random_num(low: int, high: int) -> int:
+    """Generate a random number between low and high."""
+    import random
+    result = random.randint(low, high)
+    print(result)
+    return result
+
+
+def flip_coin() -> str:
+    """Flip a coin and output heads or tails randomly."""
+    import random
+    result = 'heads' if random.randint(0, 1) == 0 else 'tails'
+    print(result)
+    return result
+
+
+def print_msg(msg: str):
+    """Print a message."""
+    print(msg)
+
+
+# source https://github.com/kubeflow/kfp-tekton/blob/master/samples/flip-coin/condition.py
+@dsl.pipeline(
+    name='conditional-execution-pipeline',
+    description='Shows how to use dsl.Condition().'
+)
+def flipcoin_pipeline():
+    flip_coin_op = components.create_component_from_func(
+        flip_coin, base_image='python:alpine3.6')
+    print_op = components.create_component_from_func(
+        print_msg, base_image='python:alpine3.6')
+    random_num_op = components.create_component_from_func(
+        random_num, base_image='python:alpine3.6')
+
+    flip = flip_coin_op()
+    with dsl.Condition(flip.output == 'heads'):
+        random_num_head = random_num_op(0, 9)
+        with dsl.Condition(random_num_head.output > 5):
+            print_op('heads and %s > 5!' % random_num_head.output)
+        with dsl.Condition(random_num_head.output <= 5):
+            print_op('heads and %s <= 5!' % random_num_head.output)
+
+    with dsl.Condition(flip.output == 'tails'):
+        random_num_tail = random_num_op(10, 19)
+        with dsl.Condition(random_num_tail.output > 15):
+            print_op('tails and %s > 15!' % random_num_tail.output)
+        with dsl.Condition(random_num_tail.output <= 15):
+            print_op('tails and %s <= 15!' % random_num_tail.output)
+# ----------------------------------- end pipeline methods -----------------------------------
+
+
+# The name of the function should be kfp_tekton_<real_fn_call>. Example:
+# real_fn_call=create_run_from_pipeline_func
+# fn=kfp_tekton_create_run_from_pipeline_func
+class DataSciencePipelinesKfpTekton:
+    def __init__(self):
+        self.client = None
+        self.api = None
+
+    def get_client(self, user, pwd, project):
+        if self.client is None:
+            self.api = DataSciencePipelinesAPI()
+            self.api.login_using_user_and_password(user, pwd, project)
+            self.client = kfp_tekton.TektonClient(
+                host=f'https://{self.api.route}/',
+                existing_token=self.api.sa_token,
+                ssl_ca_cert='/home/dlovison/github/diego-kitchen/rhods-qe-kfp-tekton/openshift.crt'
```

Comment:
This should change right? you can not have hard coded path for certificate.

Also, is it tls certificate or some other certificate and how you captured the pod certificate. I am asking this because we can automate the whole process

Reply:
Yeap. It will change

Reply:
Done